draft-ietf-quic-transport.md

```diff
@@ -1857,13 +1857,20 @@ congestion state (see {{migration-cc}}), so the port SHOULD only be changed
 infrequently.
 
 An endpoint that receives a successfully authenticated packet with a previously
-unused connection ID MUST use the next available connection ID for any packets
-it sends to that address.  To avoid changing connection IDs multiple times when
+unused connection ID MUST use a new connection ID for any future packets it
+sends to that address.  To avoid changing connection IDs multiple times when
```

Comment:
Isn't the point of this PR to cover the case you describe in the issue?  That is, you need to say "unless it is likely that the peer changed its connection ID in response to the endpoint changing."  Or something like that.  I don't know how to implement that other than to block switches when you change, unless there is a path change.  This is tricky.

Reply:
The point of the PR grew as I worked on it.  :-)  Initially, it was to replace "the next available" with "a new" now that sequence numbers are gone.  Then, while I was at it, I decided to pick up #1276 (what happens if you're out of CIDs?).  But then I ran into the fact that the existing text will lead you into that spiral.